cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/filters/PassticketFilterFactory.java

```diff
@@ -0,0 +1,84 @@
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ServerWebExchange;
+import org.zowe.apiml.cloudgatewayservice.service.InstanceInfoService;
+import org.zowe.apiml.ticket.TicketRequest;
+import org.zowe.apiml.ticket.TicketResponse;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+@Service
+public class PassticketFilterFactory extends AbstractGatewayFilterFactory<PassticketFilterFactory.Config> {
+    private final WebClient webClient;
+    private final InstanceInfoService instanceInfoService;
+    private final String ticketUrl = "%s://%s:%s/%s/api/v1/auth/ticket";
+    private final ObjectWriter writer = new ObjectMapper().writer();
+
+    public PassticketFilterFactory(WebClient webClient, InstanceInfoService instanceInfoService) {
+        super(Config.class);
+        this.webClient = webClient;
+        this.instanceInfoService = instanceInfoService;
+    }
+
+    @Override
+    public GatewayFilter apply(Config config) {
+        try {
+            final String requestBody = writer.writeValueAsString(new TicketRequest(config.getApplicationName()));
+            return (ServerWebExchange exchange, GatewayFilterChain chain) ->
+                instanceInfoService.getServiceInstance("gateway").flatMap(instances -> {
+                    for (ServiceInstance instance : instances) {
+                        return webClient.post()
+                            .uri(String.format(ticketUrl, instance.getScheme(), instance.getHost(), instance.getPort(), instance.getServiceId().toLowerCase()))
+                            .headers(headers -> headers.addAll(exchange.getRequest().getHeaders()))
+                            .bodyValue(requestBody)
+                            .retrieve()
+                            .bodyToMono(TicketResponse.class)
+                            .flatMap(response -> {
+                                String encodedCredentials = Base64.getEncoder().encodeToString((response.getUserId() + ":" + response.getTicket()).getBytes(StandardCharsets.UTF_8));
+                                final String headerValue = "Basic " + encodedCredentials;
+                                ServerHttpRequest request = exchange.getRequest().mutate().headers(headers -> {
+                                    headers.add(HttpHeaders.AUTHORIZATION, headerValue);
+                                    headers.remove(org.springframework.http.HttpHeaders.COOKIE);
+                                }).build();
+                                return chain.filter(exchange.mutate().request(request).build());
+                            });
+                    }
+                    return chain.filter(exchange);
+                });
+        } catch (JsonProcessingException e) {
+            return ((exchange, chain) -> chain.filter(exchange));
```

Comment:
Does it make sense maybe to log some message in case something like the passticket request fails?